### ISSUE TITLE
Teletext does not work when used to display subtitles

### ISSUE THREAD

**Issue by Sopor** [kodi-pvr/pvr.iptvsimple#357]
Some IPTV channels have subtitles but they are using teletext to show them and that won't work very well. First of all the on screen display will always be visible after pressing the TXT button and second, the subtitle is there but it will not change.
![image](https://user-images.githubusercontent.com/5789283/79586084-f6c6d480-80d0-11ea-8c14-9baf09ea1343.png)

When pressing the TXT button the on screen display should hide, there is a work-a-round to not show the on screen display and that is to assign a key that will enable teletext.

VLC will make it possible to choose page (what language the subtitles should show) and a nice thing is that it is possible to remove the black background and only show the text (transparency) and then there is a button for toggle subtitles on/off.
![image](https://user-images.githubusercontent.com/5789283/79585898-be26fb00-80d0-11ea-9b79-53241485fcdd.png)

So is it possible to get this to work in `PVR IPTV simple client`? At least to get the subtitle to work. The transparency is not so important even if it is a nice feature :)

**Comment by phunkyfish** [comment]
This is not a feature of the addon.

Do you have a keyboard. If I press `v` it both displays and removed teletext from the screen.

**Comment by Sopor** [comment]
As i wrote it is possibly to map a key to toggle the teletext, but the real issue is the text that won't change when it is used as subtitles. In Sweden we have channels that use the teletext to display subtitles for people with impaired hearing.

**Comment by phunkyfish** [comment]
How do you mean the text won’t change?

**Comment by phunkyfish** [comment]
So it appears with the first line of text and then is stuck like that forever?

Can you toggle it off again?

When you toggle it on again is the text different?

I’m wondering can you try this in kodi 19 (I’m assuming you are on kodi 18)? What platform are you running?

You could also try inputstream.ffmpegdirect in kodi 19 and see if that’s any different. Only if you are not running Windows though. Havn’t got teletext working on windows yet in ffmpegdirect.

**Comment by Sopor** [comment]
Yes, the first line will appear and then nothing more happens. Other channels it can work but then it is totally out of sync, 5-10 seconds ahead.

Yes, i can toggle it off.

Yes, the text is different. I also noticed that pressing OK (i assume that is the same as pressing enter) on the remote will also change the text to the next subtitle, but it will not work every time.

Yes, i'm running 18.6 (hmm, it seems that i forgot to write that) on Windows 10, but i'm also running it on a Vero 4K+.

I have installed Kodi 19 in Windows 10 and here is seems to change the subtitle but it is still 10 seconds ahead and there is a big black screen that make it unwatchable.

Kodi 19 on Windows 10:
![image](https://user-images.githubusercontent.com/5789283/79689046-98c1fa80-8252-11ea-9a74-79325f0f602d.png)

Kodi 18.6 on Windows 10:
![image](https://user-images.githubusercontent.com/5789283/79688823-33b9d500-8251-11ea-955d-a00072ef279b.png)

Kodi 18.6 on Vero 4K+:
![image](https://user-images.githubusercontent.com/5789283/79688890-a75be200-8251-11ea-9dde-70c6f7f299ae.png)

**Comment by phunkyfish** [comment]
What does the Vero4k run? Is it LibreElec?

**Comment by phunkyfish** [comment]
It appears that's OSMC. Do OSMC do a kodi 19 nightly build?

**Comment by Sopor** [comment]
I don't know if OSMC have any Kodi 19 and i'm not in a mood to install it either. That is the device i use daily to watch TV on.

**Comment by phunkyfish** [comment]
Let me see if we can get this working on ffmpegdirect on windows. Will ping you back where there is progress.

**Comment by phunkyfish** [comment]
Do you know anyone who could test this on a Mac laptop with your set up?

If it works with ffmpegdirect then at least we know adding windows support is worth it.

**Comment by Sopor** [comment]
I'll see if i can find some mac user.

**Comment by Sopor** [comment]
It was harder than i thought to find a mac user :(

**Comment by mindcook** [comment]
Same problem here in switzerland.
Teletext subtitle works (start teletext and go to the subtitle page), first line and then halts if you doesn't move mouse or some from other interface
I think it's a little problem to add automatic refresh for teletext pages.

Libreelec 9.2.4 Kodi 18.8 PVR IPTV Simple Client

**Comment by phunkyfish** [comment]
> Same problem here in switzerland.
> Teletext subtitle works (start teletext and go to the subtitle page), first line and then halts if you doesn't move mouse or some from other interface
> I think it's a little problem to add automatic refresh for teletext pages.
> 
> Libreelec 9.2.4 Kodi 18.8 PVR IPTV Simple Client

Can you confirm that it does not work with the kodi 19 alpha release?

**Comment by mindcook** [comment]
> > Same problem here in switzerland.
> > Teletext subtitle works (start teletext and go to the subtitle page), first line and then halts if you doesn't move mouse or some from other interface
> > I think it's a little problem to add automatic refresh for teletext pages.
> > Libreelec 9.2.4 Kodi 18.8 PVR IPTV Simple Client
> 
> Can you confirm that it does not work with the kodi 19 alpha release?

Same problem with Libreelec 9.2.4 Kodi 19.0 Alpha (LibreELEC-RPi2.arm-9.80-Milhouse-20200718000045-#0717-g776b589) PVR IPTV Simple Client and Enigma2 Client tested.

**Comment by phunkyfish** [comment]
Does it work if you use inputstream.ffmpegdirect on iptvsimple? It uses a different subtitle library so may work better.

Instructions on how to set the props correctly for it are here: https://github.com/xbmc/inputstream.ffmpegdirect#using-the-addon

**Comment by mindcook** [comment]
Whats the correct mimetype for rtp stream?

#KODIPROP:mimetype=

**Comment by phunkyfish** [comment]
Did you try omitting that property?

**Comment by mindcook** [comment]
Yes. No change.

**Comment by phunkyfish** [comment]
Do you seeing in your log if it’s opened by ffmpegdirect?

**Comment by padman74** [comment]
I use Kodi 19.3 on AppleTV to watch recorded .ts files from German sat TV. I also want to see subtitles that are on teletext page 149. I'm facing also a delay of >5s. I tried on Windows 10 with Kodi 18.9 with same results. Anything that I missed?

**Comment by phunkyfish** [comment]
> I use Kodi 19.3 on AppleTV to watch recorded .ts files from German sat TV. I also want to see subtitles that are on teletext page 149. I'm facing also a delay of >5s. I tried on Windows 10 with Kodi 18.9 with same results. Anything that I missed?

Can you try with and without ffmpegdirect? The mechanism used in ffmpegdirect is different to the one kodi uses.

**Comment by padman74** [comment]
The GitHub page of the ffmpegdirect add on says that subtitles are not supported on Windows. Where can I find a build for Apple  TvOS?

**Comment by phunkyfish** [comment]
It should be included in the tvos build of kodi.

**Comment by mindcook** [comment]
>  .....and second, the subtitle is there but it will not change. 
>
Solved. Add
`gui>`
`   <smartredraw>true</smartredraw>`
`   <algorithmdirtyregions>0</algorithmdirtyregions>`
`</gui>`

to advancedsettings.xml.

> When pressing the TXT button the on screen display should hide, there is a work-a-round to not show the on screen display and that is to assign a key that will enable teletext.
>
Same problem here. The OSD menu is always on in the background of the teletext subs.

**Comment by mindcook** [comment]
> Same problem here. The OSD menu is always on in the background of the teletext subs.

Problem solved by using another skin.

**Comment by phunkyfish** [comment]
> > Same problem here. The OSD menu is always on in the background of the teletext subs.
> 
> Problem solved by using another skin.

Which skin had the problem?

**Comment by mindcook** [comment]
> > > Same problem here. The OSD menu is always on in the background of the teletext subs.
> > 
> > 
> > Problem solved by using another skin.
> 
> Which skin had the problem?

OSMC 2021.08-1 default skin.

**Comment by phunkyfish** [comment]
Then please raise this problem with OSMC.

**Comment by mindcook** [comment]
> Then please raise this problem with OSMC.

-> > Problem solved by using another skin.

**Comment by phunkyfish** [comment]
If you don’t report the problem it will never get fixed.

**Comment by www2000** [comment]
I have the same issue but i like to see the teletext button in KODI if the channel have a teletext service.
